fireworks/core/firework.py

```diff
@@ -735,21 +735,21 @@ def state(self):
         m_state = 'READY'
         #states = [fw.state for fw in self.fws]
         states = self.fw_states.values()
-        leaf_states = [self.fw_states[fw_id] for fw_id in self.leaf_fw_ids]
-        if all([s == 'COMPLETED' for s in leaf_states]):
+        leaf_states = (self.fw_states[fw_id] for fw_id in self.leaf_fw_ids)
+        if all(s == 'COMPLETED' for s in leaf_states):
```

Comment:
So without the square brackets `all` can exit as soon as the first False value occurs, rather than creating the entire list and then checking items one by one